src/core/service.c

```diff
@@ -934,7 +934,7 @@ static int service_load_pid_file(Service *s, bool may_warn) {
         prio = may_warn ? LOG_INFO : LOG_DEBUG;
 
         fd = chase_symlinks(s->pid_file, NULL, CHASE_OPEN|CHASE_SAFE, NULL);
-        if (fd == -EPERM) {
+        if (fd == -ENOLINK) {
                 log_unit_full(UNIT(s), LOG_DEBUG, fd, "Permission denied while opening PID file or potentially unsafe symlink chain, will now retry with relaxed checks: %s", s->pid_file);
```

Comment:
The log message needs updating now. The first part "Permission denied while opening PID file…" doesn't really apply anymore. This is an explicit error now, hence know for sure it's about unsafe symlink transitions